apps/src/lib/kits/maker/boards/microBit/MicroBitBoard.js

```diff
@@ -96,6 +97,24 @@ export default class MicroBitBoard extends EventEmitter {
     return Promise.resolve()
       .then(() => this.openSerialPort())
       .then(serialPort => this.boardClient_.connectBoard(serialPort))
+      .then(() => {
+        // Delay for 0.25 seconds to ensure we have time to determine the firmware version.
+        return delay(250);
+      })
+      .then(() => {
+        if (
+          this.boardClient_.firmwareVersion.includes(MICROBIT_FIRMWARE_VERSION)
+        ) {
+          return Promise.resolve();
+        } else {
+          if (this.boardClient_.firmwareVersion === '') {
+            console.warn(
```

Comment:
I would also firehose log this.

Reply:
https://github.com/code-dot-org/code-dot-org/pull/47151

Reply:
Nit: I wonder if instead of `firmwareVersionUnknown`, we should go with something like `firmwareVersionTimeout`. That way we leave space for a future world where we track an outcome where the version has been returned to us, but it is not expected.